### ISSUE TITLE
chaining is broken

### ISSUE THREAD

**Issue by stefanpenner** [rwjblue/ember-computed-decorators#18]
babel fixed how initializers where so sometimes chaining decorators doesn't work correctly.

I'll sort this out, just leaving this here as a reminder.

**Comment by topaxi** [comment]
Chaining seemed to work for me in ember-cli 1.13.8 but broke by updating to 1.13.13, how is this possible? Babel and ember-computed-decorators seem to be at the same version.

I have only been using `readOnly` and `computed`.

```javascript
export default Ember.Object.extend({
  @readOnly
  @computed('someKey')
  someOtherKey(someKey) {
    return someKey
  }
})
```

Downgrading to 1.13.8 solved the problem for me.

The error I get is:
`Error: Assertion Failed: The key provided to get must be a string, you passed undefined`

I'm not sure if this directly related to this issue.

**Comment by stefanpenner** [comment]
Babel fix a big, which is what this library was written to assume. Babel changed the order, you are likely toggling between two Babel...

I am unsure if this library now correct works with the now fixed Babel.

I believe it was technically a semver break in Babel. But decorators was still an experimental feature, so no actual violation that it fixed a bug as the spec evolved

**Comment by rwjblue** [comment]
Chaining works when they are done in the correct order. Likely still something we should fix, but if you move readonly to after the computed I believe it should work.

**Comment by stefanpenner** [comment]
@rwjblue was this lib updates after Babel reversed the order decorators are played?

**Comment by rwjblue** [comment]
I thought https://github.com/rwjblue/ember-computed-decorators/pull/27 addressed that concern, but I do not know for sure.

**Comment by devinus** [comment]
Something is definitely broken right now.

Currently using `@readOnly` on my CPs is never firing them.

This gets caught by the debugger statement (the function is entered):

<img width="972" alt="screen shot 2015-12-11 at 2 08 14 am" src="https://cloud.githubusercontent.com/assets/4836/11738868/28c05872-9fac-11e5-9333-07d525c9a469.png">

This does **NOT**:

<img width="1033" alt="screen shot 2015-12-11 at 2 08 05 am" src="https://cloud.githubusercontent.com/assets/4836/11738872/31b08cf4-9fac-11e5-8000-0036f62e4941.png">

**Comment by devinus** [comment]
Putting `@readOnly` after `@computed` as @rwjblue suggests results in this error:

<img width="845" alt="screen shot 2015-12-11 at 2 10 31 am" src="https://cloud.githubusercontent.com/assets/4836/11738896/6d3cb8f6-9fac-11e5-8030-5435f6cca77f.png">

**Comment by devinus** [comment]
I'm interested in fixing this, anybody have any clues on where I could start poking around?

**Comment by stefanpenner** [comment]
I don't believe that pr fixes the order change

**Comment by topaxi** [comment]
I just tested the master branch and my issue was solved with 0042f6066ed936b81e25dd77cedfec7e299f2e95.

**Comment by devinus** [comment]
@rwjblue Can we get a new version? Seems our issues with `@readOnly` have been fixed already.

**Comment by rwjblue** [comment]
Released as v0.2.2.
